tests/jax/components/updating/checkpointer_test.py

```diff
@@ -0,0 +1,168 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import tempfile
+import time
+from dataclasses import dataclass
+from typing import Any, Dict, Optional
+
+import pytest
+from acme.jax import savers as acme_savers
+
+from mava.components.jax.updating import Checkpointer
+from mava.components.jax.updating.checkpointer import CheckpointerConfig
+from mava.core_jax import SystemParameterServer
+
+
+@dataclass
+class MockParameterStore:
+    parameters: Optional[Dict[str, Any]] = None
+    experiment_path: Optional[str] = None
+
+
+@dataclass
+class MockParameterServer:
+    store: Optional[MockParameterStore] = None
+
+
+@pytest.fixture
+def mock_parameter_server() -> MockParameterServer:
+    """Create a mock parameter server for terminator tests"""
+
+    mock_server = MockParameterServer(
+        store=MockParameterStore(
+            parameters={
+                "trainer_steps": 50,
+            },
+            experiment_path=tempfile.mkdtemp(),
+        ),
+    )
+
+    return mock_server
+
+
+@pytest.fixture
+def checkpointer() -> Checkpointer:
+    """Creates a checkpointer fixture with config
+
+    Returns:
+        ParallelSequenceAdder with ParallelSequenceAdderConfig.
+    """
+
+    checkpointer = Checkpointer(
+        config=CheckpointerConfig(checkpoint_minute_interval=1 / 60),  # type: ignore
+    )
+    return checkpointer
+
+
+def test_save_restore(
+    mock_parameter_server: SystemParameterServer,
+    checkpointer: Checkpointer,
+) -> None:
+    """Test checkpointer save and restore.
+
+    Args:
+        mock_parameter_server: Fixture SystemParameterServer.
+        checkpointer: Fixture Checkpointer.
+
+    Returns:
+        None
+    """
+    # Create checkpointer
+    checkpointer.on_parameter_server_init(server=mock_parameter_server)
+
+    system_checkpointer = mock_parameter_server.store.system_checkpointer
+
+    # Save Initial parameters and check that the file has been saved to disk
+    system_checkpointer.save()
+    saved_trainer_steps = mock_parameter_server.store.parameters["trainer_steps"]
+    assert any(
+        fname == "checkpoint"
+        for fname in os.listdir(system_checkpointer._checkpoint_dir)
+    )
+
+    # Change the parameters and check that the checkpointer has the same value
+    mock_parameter_server.store.parameters["trainer_steps"] += 50
+
+    assert (
+        system_checkpointer._checkpoint.saveable._object_to_save.state
+        == mock_parameter_server.store.parameters
+    )
+
+    # Restore the saved parameters and check that the checkpointer values are correct
+    system_checkpointer.restore()
+    assert (
+        system_checkpointer._checkpoint.saveable._object_to_save.state
```

Comment:
Should we rather not check previous parameter_server parameters with new parameter_server values and see if the changed? E.g. we update the parameters and save it. Then we change the parameters back and restore them. We then check if the parameters are the changed ones. Instead of checking the values inside checkpoint.saveable?

Reply:
This check was due to the restore happening outside the wrapper. Will edit this

Reply:
I edited the tests to check the actual parameters only in `test_save_restore` and to check that store values/ variables assignment in `test_checkpointer`

Reply:
Great thanks :)